src/adapters/GolangTypes.ts

```diff
@@ -1,22 +1,24 @@
-export type GolangTypes = IntegerTypes | FloatingTypes | Bool | Enum;
+export type GolangTypes =
+    | SignedIntegerType
+    | UnsignedIntegerType
+    | FloatingType
+    | Bool
+    | Enum;
 
-type IntegerTypes =
-    | "int8"
-    | "int16"
-    | "int32"
-    | "int64"
-    | "uint8"
-    | "uint16"
-    | "uint32"
-    | "uint64";
+export type SignedIntegerType = "int8" | "int16" | "int32" | "int64";
 
-type FloatingTypes = "float32" | "float64";
+export type UnsignedIntegerType = "uint8" | "uint16" | "uint32" | "uint64";
 
-export type NumericType = IntegerTypes | FloatingTypes;
+export type FloatingType = "float32" | "float64";
 
-export function typeIsNumeric(
+export type NumericType =
+    | SignedIntegerType
+    | UnsignedIntegerType
+    | FloatingType;
+
+export function isNumericType(
     type: string
-): type is IntegerTypes | FloatingTypes {
+): type is SignedIntegerType | UnsignedIntegerType | FloatingType {
     return (
```

Comment:
_this is not necessarily a request_

```suggestion
    return (
        isUnsignedIntegerType(type) ||
        isSignedIntegerType(type) ||
        isFloatingType(type)
    )
```

Maybe if `isUnsignedIntegerType`, `isSignedIntegerType` and `isFloatingType` took a string as an argument you could reuse them here to have less comparisons.